src/demo/App.js

```diff
@@ -17,6 +17,19 @@ const stripLocale = (pathname, locale) => {
   return pathname.replace(`/${locale}`, '');
 };
 
+class Search extends Component {
+  constructor(props) {
+    super(props);
+    this.props = props;
+  }
+
+  render() {
+    return (
+      <input placeholder={I18n.getTranslation('search', this.props)} />
+    );
+  }
+}
```

Comment:
You should be able to turn this whole thing into a [function component](https://reactjs.org/docs/components-and-props.html):

```jsx
const Search = (props) => (
  <input placeholder={I18n.getTranslation('search', props)} />
);
```